webapp/certified/views.py

```diff
@@ -0,0 +1,355 @@
+                            }
+                        )
+
+    # default to category, which contains the least specific form_factor
+    form_factor = model_release and model_release.get(
+        "form_factor", model_release["category"]
+    )
+
+    return render_template(
+        "certified/model-details.html",
+        canonical_id=canonical_id,
+        name=model_release["model"],
+        category=model_release["category"],
+        form_factor=form_factor,
+        vendor=model_release["make"],
+        major_release=model_release["certified_release"],
+        release_details=release_details,
+        has_enabled_releases=has_enabled_releases,
+        components=component_summaries,
+    )
+
+
+def certified_home():
+
+    certified_releases = api.certified_releases(limit="0")["objects"]
+    certified_makes = api.certified_makes(limit="0")["objects"]
+
+    # Laptop section
+    laptop_releases = []
+    laptop_vendors = []
+
+    # Desktop section
+    desktop_releases = []
+    desktop_vendors = []
+
+    # SoC section
+    soc_releases = []
+    soc_vendors = []
+
+    # IoT section
+    iot_releases = []
+    iot_vendors = []
+
+    # Search results filters
+    all_releases = []
+    all_vendors = []
+
+    for release in certified_releases:
+        version = release["release"]
+
+        if release not in all_releases:
+            all_releases.append(version)
+            all_releases = sorted(all_releases, reverse=True)
+
+        if int(release["laptops"]) > 0:
+            laptop_releases.append(release)
+
+        if int(release["desktops"]) > 0:
+            desktop_releases.append(release)
+
+        if int(release["smart_core"] > 1):
+            iot_releases.append(release)
+
+        if int(release["soc"] > 1):
+            soc_releases.append(release)
+
+    for vendor in certified_makes:
+        make = vendor["make"]
+
+        if make not in all_vendors:
+            all_vendors.append(make)
+            all_vendors = sorted(all_vendors)
+
+        if int(vendor["laptops"]) > 0:
+            laptop_vendors.append(vendor)
+
+        if int(vendor["desktops"]) > 0:
+            desktop_vendors.append(vendor)
+
+        if int(vendor["smart_core"] > 1):
+            iot_vendors.append(vendor)
+
+        if int(vendor["soc"] > 1):
+            soc_vendors.append(vendor)
+
+    # Server section
+    server_releases = {}
+    server_vendors = api.vendor_summaries_server()["vendors"]
+
+    for vendor in server_vendors:
+        for release in vendor["releases"]:
+            if release in server_releases:
+                server_releases[release] += vendor[release]
+            else:
+                server_releases[release] = vendor[release]
+
+    if request.args:
+        query = request.args.get("q", default=None, type=str)
+
+        # Old site replacements
+        if request.args.get("query"):
+            query = request.args.get("query", default=None, type=str)
+
```

Comment:
Instead of handling these query parameters directly, could we send a `301` redirect to tell the client the new URL scheme?

```suggestion
        # Old site replacements
        if request.args.contains(("query", "vendors")):
           args = request.args
           args.replace("query", "q")
           flask.redirect(f"/certified?{args.join(';')}")
```

Reply:
So it was a bit tricky, as the `request.args` is an `ImmutableMultiDict`, so can't change it this way. I instead created a new dict, replaced the keys, then encoded the URL. It seems to work well. And the `flask.redirect` needs a `return`